pages/admin/content-type-builder/[entity_type_slug].js

```diff
@@ -47,218 +51,299 @@ export default function ContentTypeBuilder({cache}) {
+                  type: attr[col].type,
+                  button: <AppContentBuilderButtons />,
+                })
+              : null;
+          });
+        });
 
-       } catch (ex) {
+        dispatch({ type: SET_ATTRIBUTES, payload: entries });
+      } catch (ex) {
         console.error(ex.stack);
-       } finally {
-        dispatch({type: LOADING, payload: false})
-       }
-      })();
-    }, [entity_type_slug]);
+      } finally {
+        dispatch({ type: LOADING, payload: false });
+      }
+    })();
+  }, [entity_type_slug]);
 
   const columns = [
-    { accessor: "name", displayName: "NAME", },
-    { accessor: "type", displayName: "TYPE", },
-    { accessor: "button", displayName: "", },
+    { accessor: "name", displayName: "NAME" },
+    { accessor: "type", displayName: "TYPE" },
+    { accessor: "button", displayName: "" },
   ];
 
-
   const entries = [
-      {name: <IconText name='Name' /> , type: 'Text', button: <AppContentBuilderButtons isDisabled={false} />},
-      {name: <IconNumber name='Price' /> , type: 'Number',  button: <AppContentBuilderButtons isDisabled={false} /> },
-      {name: <IconMedia name='Image1' /> , type: 'Media',  button: <AppContentBuilderButtons isDisabled={false} /> }
-  ]
+    {
+      name: <IconText name="Name" />,
+      type: "Text",
+      button: <AppContentBuilderButtons isDisabled={false} />,
+    },
+    {
+      name: <IconNumber name="Price" />,
+      type: "Number",
+      button: <AppContentBuilderButtons isDisabled={false} />,
+    },
+    {
+      name: <IconMedia name="Image1" />,
+      type: "Media",
+      button: <AppContentBuilderButtons isDisabled={false} />,
+    },
+  ];
 
   const showDeleteModal = () => {
-    dispatch({type: SHOW_DELETE_CONFIRMATION_MODAL});
+    dispatch({ type: SHOW_DELETE_CONFIRMATION_MODAL });
   };
 
   const showAddAttributeModal = () => {
-    dispatch({type: SET_SHOW, payload: true});
- };
- 
+    dispatch({ type: SET_SHOW, payload: true });
+  };
+
   const hideAddAttributeModal = () => {
-    dispatch({type: SET_SHOW, payload: false});
+    dispatch({ type: SET_SHOW, payload: false });
   };
 
-  const performDelete = ({typeSlug}) => {
+  const showEditCTypeModal = () => {
+    dispatch({ type: SHOW_EDIT_CTYPE_MODAL });
+  };
+
+  const hideEditCTypeModal = () => {
+    dispatch({ type: HIDE_EDIT_CTYPE_MODAL });
+  };
+
+  const performDelete = ({ typeSlug }) => {
     (async () => {
       console.error(`deleting... ${typeSlug}`);
-      dispatch({type: HIDE_DELETE_CONFIRMATION_MODAL});
-      await slsFetch(`/api/entity_types/${typeSlug}`,{
-        method: 'DELETE'
+      dispatch({ type: HIDE_DELETE_CONFIRMATION_MODAL });
+      await slsFetch(`/api/entity_types/${typeSlug}`, {
+        method: "DELETE",
       });
-      loadEntityTypes({rootState, rootDispatch});
+      loadEntityTypes({ rootState, rootDispatch });
     })();
   };
 
-  const onAddAnotherField = (evt) => {
+  const onFormSubmit = (evt, func = () => {}) => {
```

Comment:
Maybe renaming this variable would reveal its intention better?
```suggestion
  const onFormSubmit = (evt, callback = () => {}) => {
```